src/dbt_client/dbtCoreIntegration.ts

```diff
@@ -398,6 +426,123 @@ export class DBTCoreProjectIntegration
     this.executionInfrastructure.addCommandToQueue(command);
   }
 
+  private async getDeferParams(projectRoot: Uri): Promise<string[]> {
+    // validate credentials and if not valid run without defer params
+    const config = this.altimateRequest.getConfig();
+    if (!config?.key || !config?.instance) {
+      // only validate when both are set
+      window.showErrorMessage(
+        "Cannot run with defer to prod set up as Auth tokens are invalid",
+      );
+      throw new Error("Missing Altimate instance name and key in settings.");
+    }
+
+    const validation = await this.altimateRequest.validateCredentials(
+      config.instance,
+      config.key,
+    );
+
+    if (!validation?.ok) {
+      window.showErrorMessage(
+        "Cannot run with defer to prod set up as Auth tokens are invalid",
```

Comment:
Has this error message been reviewed by @ppradnesh and @anandgupta42 ?

Reply:
Also won't this result in mutliple error messages as we throw the error hereunder?

Reply:
@suryaiyer95

Reply:
But won't it result in an unhandled exception?